cli/server_test.go

```diff
@@ -0,0 +1,129 @@
+	e := newExecutor(t, false)
+
+	t.Run("invalid config path", func(t *testing.T) {
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("bad logger config", func(t *testing.T) {
+		badConfigDir := t.TempDir()
+		logfile := filepath.Join(badConfigDir, "logdir")
+		require.NoError(t, os.WriteFile(logfile, []byte{1, 2, 3}, os.ModePerm))
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.LogPath = filepath.Join(logfile, "file.log")
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("invalid storage", func(t *testing.T) {
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.DBConfiguration.Type = ""
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("stateroot service is on && StateRootInHeader=true", func(t *testing.T) {
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.StateRoot.Enabled = true
+			cfg.ProtocolConfiguration.StateRootInHeader = true
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("invalid Oracle config", func(t *testing.T) {
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.Oracle.Enabled = true
+			cfg.ApplicationConfiguration.Oracle.UnlockWallet.Path = "bad_orc_wallet.json"
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("invalid consensus config", func(t *testing.T) {
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.UnlockWallet.Path = "bad_consensus_wallet.json"
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("invalid Notary config", func(t *testing.T) {
+		t.Run("malformed config", func(t *testing.T) {
+			saveCfg(t, func(cfg *config.Config) {
+				cfg.ProtocolConfiguration.P2PSigExtensions = false
+				cfg.ApplicationConfiguration.P2PNotary.Enabled = true
+			})
+			e.RunWithError(t, baseCmd...)
+		})
+		t.Run("invalid wallet", func(t *testing.T) {
+			saveCfg(t, func(cfg *config.Config) {
+				cfg.ProtocolConfiguration.P2PSigExtensions = true
+				cfg.ApplicationConfiguration.P2PNotary.Enabled = true
+				cfg.ApplicationConfiguration.P2PNotary.UnlockWallet.Path = "bad_notary_wallet.json"
+			})
+			e.RunWithError(t, baseCmd...)
+		})
+	})
+	t.Run("good", func(t *testing.T) {
+		var (
+			rpcAddress string
+			rpcPort    uint16
+		)
+		saveCfg(t, func(cfg *config.Config) {
+			// firstly, get free port for RPC server
+			addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+			require.NoError(t, err)
+			l, err := net.ListenTCP("tcp", addr)
+			require.NoError(t, err)
+			defer l.Close()
+			cfg.ApplicationConfiguration.RPC.Port = uint16(l.Addr().(*net.TCPAddr).Port)
+			cfg.ApplicationConfiguration.RPC.Enabled = true
+			rpcAddress = cfg.ApplicationConfiguration.RPC.Address
+			rpcPort = cfg.ApplicationConfiguration.RPC.Port
+		})
+		go func() {
+			e.Run(t, baseCmd...)
+		}()
+
+		gctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+		c, err := client.New(gctx, "http://"+rpcAddress+":"+strconv.FormatUint(uint64(rpcPort), 10), client.Options{})
```

Comment:
But you need to have the port here. Maybe we can just check the node log?

Reply:
Fixed.